### DISCUSSION TITLE
Website to single PDF file

### DISCUSSION THREAD

**Discussion by emil-peters** [quarto-dev/quarto-cli#4274] [Q&A] [answered]
Hi there,

Using `quarto preview . --to pdf` I can generate PDF files for all qmd files.
My end goal is that all of these should be merged into a single PDF.

I also have tried using a second profile, and transforming my website into a book. There I can generate a single PDF, but the issue I then get is that all my website's listing are no longer displayed. So all these pages are missing from the PDF.

What would be the best solution to this issue?

**Comment by rgaiacs** [comment, marked as answer]
Dear @emil-peters, would a post process work for you? For example, you can merge the PDF files for all `.qmd` files using [`pdftk`](https://www.pdflabs.com/tools/pdftk-the-pdf-toolkit/).

**Reply by mcanouil** [reply]
Indeed, this is probably the easiest path.
Otherwise "complex" profiles using different formats will be required and for that it's highly dependent on what's your goal and what are you files.

**Comment by emil-peters** [comment]
Thanks for the feedback, we are generating `qmd` files through python containing huge amount of images and pages (~1500  pages).
It would be a lot of work to combine all these things ourselves.

We decided to go with the approach of just providing the website (generated html), and generating a small single page PDF containing the most important things. As sort of an overview.

If in the future it would be possible to generate a fully linked PDF based on a website and all it's links, please let me know.

**Reply by mcanouil** [reply]
I think it's unlikely à feature to turn a website to a pdf will ever happen.

Either, setup a profile with another format and render both, and use post-render feature (as suggested) from Quarto to post process the generated files.
